### PR TITLE
SONARIAC-1357 Raise Kubernetes issues on yaml values instead of key-value pairs

### PR DESCRIPTION
**Pull request by mstachniuk**

### CODE REVIEW

File: iac-common/src/main/java/org/sonar/iac/common/yaml/object/ListObject.java [hunk 1]
```diff
@@ -62,13 +65,17 @@ public Stream<YamlTree> getItemIf(Predicate<YamlTree> predicate) {
   }
 
   public ListObject reportIfAnyItem(Predicate<YamlTree> predicate, String message) {
-    getItemIf(predicate).findFirst().ifPresent(item -> report(message));
+    getItemIf(predicate).findFirst().ifPresent(item -> reportOnItems(message));
```
Comment (petertrr) [inline iac-common/src/main/java/org/sonar/iac/common/yaml/object/ListObject.java:68]: Shouldn't it highlight only on matching items then?

File: iac-extensions/kubernetes/src/test/resources/checks/CapabilitiesCheck/CapabilitiesChart/templates/capabilities-pod.yaml [hunk 2]
```diff
@@ -1,33 +1,4 @@
-apiVersion: v1
```
Comment (petertrr) [inline iac-extensions/kubernetes/src/test/resources/checks/CapabilitiesCheck/CapabilitiesChart/templates/capabilities-pod.yaml:1]: Why did you remove these tests?

File: iac-extensions/kubernetes/src/test/resources/checks/CapabilitiesCheck/CapabilitiesChart/templates/capabilities-pod.yaml [hunk 3]
```diff
@@ -41,9 +12,8 @@ spec:
       {{- if .Values.capabilities }}
         capabilities:
           add:
-# Noncompliant@+1
+# Noncompliant@+2
           {{- range .Values.capabilities }}
-#                          ^^^^^^^^^^^^^^^^
             - {{ . | quote }}
```
Comment (petertrr) [inline iac-extensions/kubernetes/src/test/resources/checks/CapabilitiesCheck/CapabilitiesChart/templates/capabilities-pod.yaml:17]:
```suggestion
            - {{ . | quote }}
            #^[sc=0;ec=29]
```
So now we will highlight this part, right?

**Review by petertrr — APPROVED** [review]
LGTM. That's a lot of updated tests, good job!

File: iac-extensions/kubernetes/src/test/resources/checks/CapabilitiesCheck/CapabilitiesChart/templates/capabilities-pod.yaml (hunk 2 above)
Comment (mstachniuk) [reply iac-extensions/kubernetes/src/test/resources/checks/CapabilitiesCheck/CapabilitiesChart/templates/capabilities-pod.yaml:1]: Good catch. Removed because I debugged the last case and forgot to restore it. Now it's resorted.

File: iac-extensions/kubernetes/src/test/resources/checks/CapabilitiesCheck/CapabilitiesChart/templates/capabilities-pod.yaml (hunk 3 above)
Comment (mstachniuk) [reply iac-extensions/kubernetes/src/test/resources/checks/CapabilitiesCheck/CapabilitiesChart/templates/capabilities-pod.yaml:17]: Right. But I'm not a fan of this syntax and would have a clean loop without comments in between.
Unfortunately, this change introduced regression in such Helm loops, because now the issue is raised on `{{ . | quote }}` what doesn't contain any value path and the trailing comment with line numbers doesn't include the above line.
We have to think about how to solve such cases (detecting loops, looking before, etc.) but it's a topic for the next iteration.

File: iac-common/src/main/java/org/sonar/iac/common/yaml/object/ListObject.java (hunk 1 above)
Comment (mstachniuk) [reply iac-common/src/main/java/org/sonar/iac/common/yaml/object/ListObject.java:68]: The current logic was that it matches all items, e.g. all Capabilities are sensitive. I guess such a method may be needed in the future.
